src/Proxy.NET/Routes/AzureOpenAI.cs

```diff
@@ -0,0 +1,99 @@
+using System.Net;
+using System.Text.Json;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Proxy.NET.Authentication;
+using Proxy.NET.Models;
+using Proxy.NET.Services;
+
+namespace Proxy.NET.Routes;
+
+public static class AzureAI
+{
+    public static RouteGroupBuilder MapAzureOpenAIRoutes(this RouteGroupBuilder builder)
+    {
+        var openAIGroup = builder.MapGroup("/openai/deployments/{deploymentName}");
+        openAIGroup.MapPost("/chat/completions", ProcessRequestAsync);
+        openAIGroup.MapPost("/extensions/chat/completions", ProcessRequestAsync);
+        openAIGroup.MapPost("/completions", ProcessRequestAsync);
+        openAIGroup.MapPost("/embeddings", ProcessRequestAsync);
+        openAIGroup.MapPost("/images/generations", ProcessRequestAsync);
+        return builder;
+    }
+
+    [Authorize(AuthenticationSchemes = ProxyAuthenticationOptions.ApiKeyScheme)]
+    private static async Task<IResult> ProcessRequestAsync(
+        [FromServices] ICatalogService catalogService,
+        [FromServices] IProxyService proxyService,
+        [FromQuery(Name = "api-version")] string apiVersion,
+        [FromBody] JsonDocument requestJsonDoc,
+        HttpContext context,
+        string deploymentName
+    )
+    {
+        using (requestJsonDoc)
+        {
+            var routePattern = (context.GetEndpoint() as RouteEndpoint)?.RoutePattern.RawText;
+            var extPath = routePattern?.Split("{deploymentName}").Last();
+            var requestContext = (RequestContext)context.Items["RequestContext"]!;
+
+            var streaming = IsStreaming(requestJsonDoc);
+            var maxTokens = GetMaxTokens(requestJsonDoc);
+
+            requestContext.DeploymentName = deploymentName;
+
+            if (maxTokens.HasValue && maxTokens > requestContext.MaxTokenCap && requestContext.MaxTokenCap > 0)
+            {
+                throw new HttpRequestException(
+                    $"max_tokens exceeds the event max token cap of {requestContext.MaxTokenCap}",
+                    null,
+                    HttpStatusCode.BadRequest
+                );
+            }
+
+            var deployment = await catalogService.GetCatalogItemAsync(requestContext);
+            var url = GenerateEndpointUrl(deployment, extPath!, apiVersion);
+
+            if (streaming)
+            {
+                await proxyService.HttpPostStreamAsync(url, deployment.EndpointKey, context, requestJsonDoc, requestContext);
+                return new ProxyResult(null!, (int)HttpStatusCode.OK);
+            }
+            else
```

Comment:
Don't need to wrap with an `else` here, as the `return` on line 60 will mean there's no continuation, so you can avoid the nesting.